src/solvers/smt2/smt2_conv.cpp

```diff
@@ -3825,11 +3825,23 @@ void smt2_convt::convert_minus(const minus_exprt &expr)
        expr.op1().type().id()==ID_pointer)
     {
       // Pointer difference
-      auto element_size =
-        pointer_offset_size(to_pointer_type(expr.op0().type()).base_type(), ns);
-      CHECK_RETURN(element_size.has_value() && *element_size >= 1);
+      const auto &base_type = to_pointer_type(expr.op0().type()).base_type();
+      mp_integer element_size;
```

Comment:
:thinking: It might be nice if this code which works out the `element_size` were to be factored out into a separate function. This commit seems to be duplicating the code from pointer addition.